check/README.md

```diff
@@ -0,0 +1,99 @@
+# Tool for work with string tables
+
+## Glossary
+
+- `string table` a file with translation
+- `reference table` en.txt string table.
+  English is the main game language, so any changes related to text should be in that file.
+  Also this table us used as fallback for other translation.
+  This file is mostly maintained by people who change game code,
+  so the list of entries in this file should be the most actual.
+- `entry` one piece of translation, it consists of
+  - `key` a sting that represents entry in the codebase (C++, Python).
+    This string must be unique.
+  - `value` actual translation that will be shown to the user.
+
+
+### Key
+Only English chars in uppercase and dash are allowed.
+Rename of a key should be done with caution. Usually key used as is in code,
+but sometimes it could be used as prefix and code logic based on name could be present.
+
+### Value syntax
+#### Simple (single line)
+A single line of text, should not be started and ended by any whitespace character.
+
+```
+KEY
+value
+
+>>>
+value
+<<<
+```
+
+#### Full (single or multiline)
+Start and end with triple single quotes.
+You can put anything except triple singe quotes inside.
+
+```
+KEY_WITH_LEADING_SPACE
+''' value'''
+
+>>>
+ value
+<<<
+
+KEY_WITH_MULTILINE
+'''Hello
+world'''
+
+>>>
+Hello
+world
+<<<
+
+```
+
+
+#### Tags
+You can customize text with special markup tags. Not full list of tags:
+- `reference` special syntax that allows to put value from another entry.
+  You add a key in double square braces and user will see the value of that key.
```

Comment:
This is misleading. `[[KEY]]` does a substitution, and if there is a stringtable entry with the key `KEY`, it will be simply replaced. But those substitutions are recursive, so if there is a stringtable entry
````
KEY
Some text [[ANOTHER_KEY]] and perhaps a [[tech LRN_PSYONICS]]
````
Then the references and link generation in the stringtable entry with key `KEY` will be resolved first, then substituted into the place where `[[KEY]]` was written.

Reply:
Is it possible to create a recursive cycle?   do we have a recursion limit?

Reply:
There is [a check for recursion](https://github.com/freeorion/freeorion/blob/master/util/StringTable.cpp#L247) that will generate an error message or abort when it finds one.